blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodCounterRollupTest.java

```diff
@@ -1,93 +1,758 @@
+
+    @Test
+    public void withCountFloatSetsCount() {
+        // given
+        BluefloodCounterRollup rollup = new BluefloodCounterRollup();
+        // precondition
+        assertNull(rollup.getCount());
+        // when
+        rollup.withCount(123.45f);
+        // then
+        Number value = rollup.getCount();
+        assertThat(value, is(CoreMatchers.<Number>instanceOf(Double.class)));
+        assertEquals(123.45d, value.doubleValue(), 0.00001d);
+    }
+
+    @Test
+    public void withCountDoubleSetsCount() {
+        // given
+        BluefloodCounterRollup rollup = new BluefloodCounterRollup();
+        // precondition
+        assertNull(rollup.getCount());
+        // when
+        rollup.withCount(123.45d);
+        // then
+        Number value = rollup.getCount();
+        assertThat(value, is(CoreMatchers.<Number>instanceOf(Double.class)));
+        assertEquals(123.45d, value.doubleValue(), 0.00001d);
+    }
+
+    @Test
+    public void withRateSetsRate() {
+        // given
+        BluefloodCounterRollup rollup = new BluefloodCounterRollup();
+        // precondition
+        assertEquals(0.0d, rollup.getRate(), 0.00001d);
+        // when
+        rollup.withRate(123.45d);
+        // then
+        assertEquals(123.45d, rollup.getRate(), 0.00001d);
+    }
+
+    @Test
+    public void withSampleCountSetsSampleCount() {
+        // given
+        BluefloodCounterRollup rollup = new BluefloodCounterRollup();
+        // precondition
+        assertEquals(0, rollup.getSampleCount());
+        // when
+        rollup.withSampleCount(123);
+        // then
+        assertEquals(123, rollup.getSampleCount());
+    }
+
+    @Test
+    public void rawSampleBuilderWithEmptyInputYieldsAllZeroes() throws IOException {
+        // given
+        Points<SimpleNumber> input = new Points<SimpleNumber>();
+        // when
+        BluefloodCounterRollup rollup = BluefloodCounterRollup.buildRollupFromRawSamples(input);
+        // then
+        Number count = rollup.getCount();
+        assertNotNull(count);
+        assertThat(count, is(CoreMatchers.<Number>instanceOf(Long.class)));
+        assertEquals(0L, count.longValue());
+        assertEquals(0.0d, rollup.getRate(), 0.00001d);
+        assertEquals(0, rollup.getSampleCount());
+    }
+
+    @Test
+    public void rawSampleBuilderWithSingleInputYieldsThatInputWithInfiniteRate() throws IOException {
+        // given
+        Points<SimpleNumber> input = new Points<SimpleNumber>();
+        input.add(new Points.Point<SimpleNumber>(1234L, new SimpleNumber(3L)));
+        // when
+        BluefloodCounterRollup rollup = BluefloodCounterRollup.buildRollupFromRawSamples(input);
+        // then
+        Number count = rollup.getCount();
+        assertNotNull(count);
+        assertThat(count, is(CoreMatchers.<Number>instanceOf(Long.class)));
+        assertEquals(3L, count.longValue());
+        assertEquals(Double.POSITIVE_INFINITY, rollup.getRate(), 0.00001d); //(3)/(1-1)=inf
```

Comment:
@VinnyQ Yes, you can test for infinity, but I'm not sure if this is the best way to do so. Floating point arithmetic is notoriously tricky, with plenty of special cases and rules. Now that I think about it, this should be replaced with `assertTrue(Double.isInfinite(rollup.getRate())`. `getRate()` is returning infinity because that's what was set as the rate by `buildRollupFromRawSamples`. `buildRollupFromRawSamples` divides a non-zero number by zero, the result of which is infinite, under floating point math rules.